### DISCUSSION TITLE
自动生成代码片段支持

### DISCUSSION THREAD

**Discussion by zzx31584729** [helix-editor/helix#9480] [General] [answered]
我没有找到关于生成代码片段的功能，like：
err := a_golang_func()
if err != nil {
    return err
}
当我输入iferr时会给我提示
if err!=nil{ return err }

**Comment by TornaxO7** [comment]
Translation (with [deepL](https://www.deepl.com/translator)):
```
I didn't find a function about generating code snippets, like:
err := a_golang_func()
if err ! = nil {
return err
}
When I type iferr it gives me the prompt
if err != nil { return err }
```

**Comment by TornaxO7** [comment, marked as answer]
@zzx31584729 could you please elaborate what you mean with
> (..) generating code snippets (...)

? Did you expect that the auto-completion suggested your provided code-snippet or are you asking if there's a way to implement custom code snippets (https://github.com/helix-editor/helix/issues/395)?

Also please write your messages in english so that there are potentially more people who could help you.

**Comment by Dangerise** [comment]
@TornaxO7 I read his words, and he wants a code snippet like that in Go

**Comment by TornaxO7** [comment]
> @TornaxO7 I read his words, and he wants a code snippet like that in Go

nice, thank you for the better translation! :) 

I think in this case he'd have to wait until https://github.com/helix-editor/helix/issues/395 is completed.

**Comment by pascalkuthe** [comment]
Opening an issue on an obviously English project in a other language is pretty tactless

**Comment by zzx31584729** [comment]
Thank you very much.
